### DISCUSSION TITLE
custom colormaps for raster data

### DISCUSSION THREAD

**Discussion by Paree24** [opengeos/leafmap#506] [Q&A] [answered]
Hello,

I wanted to plot local AQI raster data in leafmap and use these colours
https://www.airnow.gov/aqi/aqi-basics/

I created a matplotlib cmap but it looks like the library only supports built in cmaps. Is there a way to use custom colors for single band raster data?

I also wanted to use split map to see different in AQI data from current hour to next hour but from what i can get from the documentation, split map either needs a pre existing basemap or COGs and doesn't work eith local raster files.

Is there any workaround for this or can this be added as a feature? (Especially custom matplotlib cmap objects)

**Comment by giswqs** [comment, marked as answer]
Good question. Here is an example. 

```python
m.add_raster(dem, cmap=['#ff0000', '#ffff00', '#0000ff'], layer_name="DEM")
```
![image](https://github.com/opengeos/leafmap/assets/5016453/b225cee9-8b0f-4c2b-93aa-82c012edc32b)

**Comment by Paree24** [comment]
Thanks
